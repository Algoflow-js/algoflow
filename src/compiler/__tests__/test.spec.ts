import { describe, expect, it } from "vitest";
import { stepify } from "../index";

describe("test", () => {
  it("should pass", () => {
    const code = `function twoNumberSum(array, targetSum) {
        const hash = {}
        for(let number of array){
            if(hash[number]){
                return number > hash[number] ? [hash[number], number] : [number, hash[number]]
            }
            hash[targetSum - number] = number;
        }
        return []
    }
    `;
    console.log(stepify(code));
  });
});
