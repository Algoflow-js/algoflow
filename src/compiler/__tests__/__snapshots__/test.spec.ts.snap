// Vitest Snapshot v1

exports[`test > should pass 1`] = `
"const twoNumberSum = (array, targetSum) => {
  cache.__(array, \\"array\\");

  cache.__(targetSum, \\"targetSum\\");

  const hash = cache._o({}, \\"hash\\");

  for (let number of array) {
    cache.__(number, \\"number\\");

    if (hash[number]) {
      return cache.__(number > hash[number] ? [hash[number], number] : [number, hash[number]], \\"number > hash[number] ? [hash[number], number] : [number, hash[number]]\\");
    }

    cache._0 = cache.__(targetSum - number, \\"targetSum - number\\")
    hash[cache._0] = cache.__(number, \\"hash[targetSum - number]\\");
  }

  return cache._a([], \\"[]\\");
};"
`;
